dist/formatters/no-internal-summary-table-creator.js

```diff
@@ -0,0 +1,126 @@
+      const newCount = currentCount + 1;
+      errorTracker.set(key, newCount);
+
+      const tagViolationCount = tagViolationsTracker.get(tag) ?? 0;
+      tagViolationsTracker.set(tag, tagViolationCount + 1);
+    }
+  });
+
+    if (problemFiles.size === 0 || errorTracker.size === 0)
+      return;
+
+    // iterate over the problemFiles and for each value, remove the last comma from locations
+    for (const [key, value] of problemFiles.entries()) {
+      const locations = value.locations.slice(0, -1);
+      problemFiles.set(key, { locations, count: value.count});
+    }
+
+    const filesSummaryRows = [
+      ['File', 'Locations', 'Tag', '# of Occurrences'],
+      ...Array.from(problemFiles.entries()).map(([filePath, { locations, count }]) => {
+        const [relativeFilePath, tag] = filePath.split(';;');
+        return [relativeFilePath, locations, tag, count];
+      })
+    ];
+    const filesSummaryTable = createAsciiTable(filesSummaryRows, maxCellWidth);
+
+    const rows = [
+      ["Kind and Name", "Tag", "# of Occurrences"],
+      ...Array.from(errorTracker.entries()).map(([key, value]) => {
+        const [kind, name, tag] = key.split(";;");
+        return [`${kind} ${name}`, tag, value];
+      })
+
+    ];
+    const summaryTable = createAsciiTable(rows, maxCellWidth);
+
+    const tagViolationsRows = [
+      ["Tag", "# of Occurrences"],
+      ...Array.from(tagViolationsTracker.entries()).map(([tag, value]) => [tag, value]),
+    ];
+    const tagViolationsTable = createAsciiTable(tagViolationsRows, maxCellWidth);
+    // create a csv of the 2D arrays problemFiles and errorTracker and save that csv file in the current working directory
+    // Convert 2D arrays to CSV format
+    const problemFilesCSV = Array.from(problemFiles.entries()).map(([filePath, { locations, count }]) => {
+      const [relativeFilePath, tag] = filePath.split(';;');
+      return `${relativeFilePath},"${locations}",${tag},${count}`;
+    }).join('\n');
+    const errorTrackerCSV = Array.from(errorTracker.entries()).map(([key, value]) => {
+      const [kind, name, tag] = key.split(";;");
+      return `${kind} ${name},${tag},${value}`;
+    }).join('\n');
+    const tagViolationsCSV = Array.from(tagViolationsTracker.entries()).map(([tag, value]) => `${tag},${value}`).join('\n');
+
+    const allTablesSummaryTitle = `Summary tables for '${ruleId}' lint rule violations:`;
+    const problemFilesTitle = `'${ruleId}' violations summary table by files:`;
+    const errorTrackerTitle = `'${ruleId}' violations summary table by kind and name:`;
+    const tagViolationsTitle = `'${ruleId}' violations summary table by tags:`;
+
+    // Combine problemFilesCSV and errorTrackerCSV
+    const combinedCSV = `${allTablesSummaryTitle}\n\n${errorTrackerTitle}\nKind and Name,Tag,# of Occurrences\n${errorTrackerCSV}\n\n${problemFilesTitle}\nFile,Locations,Tags\n${problemFilesCSV}\n\n${tagViolationsTitle}\nTag,# of Occurrences\n${tagViolationsCSV}\n`;
```

Comment:
Using newlines instead of commas for the CSV delimiter?

Reply:
no, the new lines is to separate the titles and the tables from one another

Reply:
generated example csv:
<img width="845" alt="image" src="https://github.com/iTwin/eslint-plugin/assets/98850418/345d8841-c787-4ee1-9e56-e1a2d0d76804">
<img width="1208" alt="image" src="https://github.com/iTwin/eslint-plugin/assets/98850418/5e240986-ab9e-406a-857e-9233900cd6b9">

Reply:
Ah indeed. Thanks

Reply:
imo let's stick with one csv, if the consumer wants to split it they can do so manually, or if the teams want us to implement this we can also do so if there is demand but for now I would go with just one collated csv